utils/redisIO.py

```diff
@@ -5,101 +5,190 @@
         data = json.dumps(data, **kwargs)
-        return self.set(key, data)
+        return await self.set(key, data)
 
-    def not_json_get(self, key, default=None):
-        data = self.get(key)
+    async def not_json_get(self, key, default=None):
+        data = await self.get(key)
         return json.loads(data) if data is not None else default
 
-    def publish(self, channel, data):
-        return self._db.publish(channel, data)
+    # ==== lists ====
+    async def llen(self, key):
+        return await self._db.llen(key)
 
-    def hget(self, key, field, default=None):
-        encoded_data = self._db.hget(key, field)
-        return encoded_data.decode() if encoded_data is not None else default
+    async def lindex(self, key, index):
+        encoded_data = await self._db.lindex(key, index)
+        return encoded_data.decode() if encoded_data is not None else None
 
-    def hset(self, key, field, value):
-        return self._db.hset(key, field, value)
+    async def rpush(self, key, *values):
+        return await self._db.rpush(key, *values)
 
-    def hdel(self, key, *fields):
-        return self._db.hdel(key, *fields)
+    # ==== pubsub ====
+    async def subscribe(self, *channels):
+        return await self._db.subscribe(*channels)
 
-    def jhget(self, key, field, default=None):
-        data = self.hget(key, field)
-        return json.loads(data) if data is not None else default
+    async def publish(self, channel, data):
+        return await self._db.publish(channel, data)
 
-    def jhset(self, key, field, value, **kwargs):
-        data = json.dumps(value, **kwargs)
-        return self.hset(key, field, data)
+
+class _PubSubMessageBase:
+    def __init__(self, type, id, sender):
+        self.type = type
+        self.id = id
+        self.sender = sender
+
+    @classmethod
+    def from_dict(cls, d):
+        return cls(**d)
+
+    def to_dict(self):
+        return {"type": self.type, "id": self.id, "sender": self.sender}
+
+    def to_json(self):
+        return json.dumps(self.to_dict())
+
+
+class PubSubCommand(_PubSubMessageBase):
+    def __init__(self, id, sender, command, args, kwargs):
+        super(PubSubCommand, self).__init__('cmd', id, sender)
+        self.command = command
+        self.args = args
+        self.kwargs = kwargs
+
+    @classmethod
+    def new(cls, bot, command, args=None, kwargs=None):
+        if args is None:
+            args = []
+        if kwargs is None:
+            kwargs = {}
+        _id = str(uuid.uuid4())
+        return cls(_id, bot.cluster_id, command, args, kwargs)
+
+    def to_dict(self):
+        inst = super(PubSubCommand, self).to_dict()
+        inst.update({"command": self.command, "args": self.args, "kwargs": self.kwargs})
+        return inst
+
+
+class PubSubReply(_PubSubMessageBase):
+    def __init__(self, id, sender, reply_to, data):
+        super(PubSubReply, self).__init__('reply', id, sender)
```

Comment:
python 3 doesn't require passing in the class for super()
```suggestion
        super().__init__('reply', id, sender)
```